dotnet/src/SemanticKernel/SemanticFunctions/PromptFunction.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.AI;
+using Microsoft.SemanticKernel.AI.TextCompletion;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.SemanticFunctions;
+
+#pragma warning disable format
+
+/// <summary>
+/// A Semantic Kernel "Semantic" prompt function.
```

Comment:
What about these prompt functions are semantic?

Reply:
Real question?

Internal class & implementation - if the comment is confusing, we can just remove it.

Reply:
I just wanted to make sure we weren't limiting what the results|output of a prompt function could be. 

not a blocker for pr.